### PR TITLE
CSV-278: Reuse Buffers in Lexer for Delimiter Detection

### PR DESCRIPTION
**Pull request by belugabehr**
https://issues.apache.org/jira/browse/CSV-278

### CODE REVIEW

File: src/main/java/org/apache/commons/csv/ExtendedBufferedReader.java [hunk 1]
```diff
@@ -132,10 +132,24 @@ int lookAhead() throws IOException {
      */
     char[] lookAhead(final int n) throws IOException {
         final char[] buf = new char[n];
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/csv/ExtendedBufferedReader.java:134]: The local var is not needed anymore right?

File: src/main/java/org/apache/commons/csv/Lexer.java [hunk 2]
```diff
@@ -112,25 +116,24 @@ boolean isCommentStart(final int ch) {
     }
 
     /**
-     * Determine whether the next characters constitute a delimiter through {@link ExtendedBufferedReader#lookAhead(int)}
+     * Determine whether the next characters constitute a delimiter through
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/csv/Lexer.java:119]: No need to change for formatted, it makes more noise to review.

File: src/main/java/org/apache/commons/csv/Lexer.java [hunk 3]
```diff
@@ -153,25 +156,26 @@ boolean isEscape(final int ch) {
     }
 
     /**
-     * Tests if the next characters constitute a escape delimiter through {@link ExtendedBufferedReader#lookAhead(int)}.
+     * Tests if the next characters constitute a escape delimiter through
+     * {@link ExtendedBufferedReader#lookAhead(char[])}.
      *
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/csv/Lexer.java:161]: Same comment.

**Review by garydgregory — CHANGES_REQUESTED** [review]
Thanks for teh the PR, please see comments.

File: src/main/java/org/apache/commons/csv/ExtendedBufferedReader.java (hunk 1 above)
Comment (belugabehr) [reply src/main/java/org/apache/commons/csv/ExtendedBufferedReader.java:134]: Actually, it's still used in `CSVFormat#printWithEscapes`.  I am happy to remove fully in another PR, but wanted to keep change as small as possible.

**Comment by garydgregory** [comment]
This PR has no description or URL to the Jira ticket.

**Comment by garydgregory** [comment]
This PR has no description or URL to the Jira ticket.

**Comment by garydgregory** [comment]
@belugabehr 
May you please resolve conflicts and post your JMH benchmark results?
TY!

**Comment by belugabehr** [comment]
@garydgregory Thanks so much.  I've got one more change to propose after this then I'll share with you perf numbers between master and previous release.  Thanks!!

**Comment by garydgregory** [comment]
@belugabehr 
Please provide JMH numbers for this PR if you are providing this for performance reasons. Otherwise, it's just  a change to complex code not backed up by anything. 
If you want to put more in this PR that is fine as well.
TY

**Comment by belugabehr** [comment]
@garydgregory I posted some local results to the comment section of the JIRA.  Thanks!

https://issues.apache.org/jira/browse/CSV-278

**Comment by belugabehr** [comment]
@garydgregory Have you had a chance to check out the JMH stuff?  Thanks!

https://issues.apache.org/jira/browse/CSV-278

**Comment by garydgregory** [comment]
Maybe tonight or this weekend.

**Pull request merged**